[torchelastic][rendezvous] Add option to enable libuv for TCPStore based rendezvous backend

kurman opened the pull request:
Summary:
Expose an option to enable libuv in TCPStore based rendezvous backend that will allow better scaling.

Libuv support has been added recently and allows scaling for more than 2K nodes.

Test Plan: Unit tests

Differential Revision: D53335860




cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

wconstab commented on test/distributed/elastic/rendezvous/c10d_rendezvous_backend_test.py:
```diff
@@ -197,6 +197,18 @@ def test_create_backend_returns_backend_if_read_timeout_is_not_specified(self) -
+
+        self._assert_create_backend_returns_backend()
+
+    def test_create_backend_returns_backend_with_explicitly_without_libuv(self) -> None:
```
nit 'with explicitly without'

kurman commented:
@pytorchbot merge

kurman committed:
[torchelastic][rendezvous] Add option to enable libuv for TCPStore based rendezvous backend (#118944)

Summary:

Expose an option to enable libuv in TCPStore based rendezvous backend that will allow better scaling.

Libuv support has been added recently and allows scaling for more than 2K nodes.

Test Plan: Unit tests

Differential Revision: D53335860

kurman commented:
@pytorchbot merge
